lightning/src/chain/mod.rs

```diff
@@ -69,7 +69,7 @@ pub trait Access: Send + Sync {
 /// [`PermanentFailure`]: channelmonitor/enum.ChannelMonitorUpdateErr.html#variant.PermanentFailure
 pub trait Watch: Send + Sync {
 	/// Keys needed by monitors for creating and signing transactions.
-	type Keys: ChannelKeys;
+	type ChanSigner: Sign;
```

Comment:
Similarly here only for using an associated type vs function type parameter.

Reply:
Hmm, so this doesn't make sense as a function parameter - users don't ever want to receive arbitrary `ChannelMonitor`s with unknown signer while serializing. That said, this probably shouldn't be an associated type - a `Watch<A>` is a different type from a `Watch<B>` and a single struct could (though probably won't) implement both.